File: solutions/go/grains/1/grains.go
package grains

import(
    "errors"
)

func Square(number int) (uint64, error) {
	if number<=0 || number>64{
        return 0,errors.New("Invalid square number.")
    }
    var a uint64 = 0
    i := 1
    for i <=number{
        if a==0{
            a = 1
        }else{
            a = a * 2
        }
        i++
    }
    return a, nil
}

func Total() uint64 {
	var s uint64 = 0
    i :=1
    for i<=64{
        v,_ := Square(i)
        s += v
        i++
    }
    return s
}

File: solutions/c/grains/1/grains.c
#include "grains.h"

uint64_t square(uint8_t n){
    if (n==0){
        return 0;
    }
    uint64_t grain = 1;
    uint64_t i =1;
    while (i<n){
        grain *= 2;
        i++;
    }
    return grain;
}

uint64_t total(){
    uint64_t sum = 1;
    uint64_t grain = 1;
    uint64_t i = 1;
    while (i<=64){
        grain *= 2;
        sum += grain;
        i++;
    }
    return sum;
}